app/demo-loop/demo-loop.cc

```diff
@@ -62,6 +62,106 @@ using namespace celeritas;
 
 namespace
 {
+//---------------------------------------------------------------------------//
+/*!
+ *`RootStepWriterFilter` helper functions.
+ */
+bool rsw_filter_match(size_type step_attr_id, size_type filter_id)
+{
+    return filter_id == MCTruthFilter::unspecified()
+           || step_attr_id == filter_id;
+}
+
+bool rsw_filter_match(size_type step_trk_id, std::vector<size_type> vec_trk_id)
```

Comment:
Common mistake of missing a `const &` for passing parameters: if it's a nontrivial object that you're passing, use a const reference.
```suggestion
bool rsw_filter_match(size_type step_trk_id, std::vector<size_type> const& vec_trk_id)
```